Tests/SentryTests/SentryCrash/SentryStacktraceBuilderTests.swift

```diff
@@ -68,6 +68,32 @@ class SentryStacktraceBuilderTests: XCTestCase {
         XCTAssertTrue(filteredFrames.count == 1, "The frames must be ordered from caller to callee, or oldest to youngest.")
     }
      
+    @available(macOS 12.0, iOS 15.0, watchOS 8.0, tvOS 15.0, *)
+    func testConcurrentStacktraces() async throws {
+        guard #available(macOS 12.0, iOS 15.0, watchOS 8.0, tvOS 15.0, *) else { return }
+        SentrySDK.start { options in
+            options.dsn = TestConstants.dsnAsString(username: "SentryStacktraceBuilderTests")
+        }
+
+        await Task { await innerFrame1() }.value
+    }
+
+    @available(macOS 12.0, iOS 15.0, watchOS 8.0, tvOS 15.0, *)
+    func innerFrame1() async {
+        await Task { @MainActor in }.value
+        await innerFrame2()
+    }
+
+    @available(macOS 12.0, iOS 15.0, watchOS 8.0, tvOS 15.0, *)
+    func innerFrame2() async {
+        let needed = ["testConcurrentStacktraces", "innerFrame1", "innerFrame2"]
+        let actual = fixture.sut.buildStacktraceForCurrentThreadAsyncUnsafe()!
```

Comment:
It's worth mentioning that I had to use the `AsyncUnsafe` variant of this method because the other variant uses `sentrycrashdl_dladdr` which has trouble finding the symbol names (whereas `dladdr` works fine). Though I suspect that might be orthogonal to this PR.